app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -63,12 +63,24 @@ impl ModuleGroups {
 ///
 /// The builder allow extending the list with new entries, and build a list with properly sorted
 /// groups.
+///
+/// To build a [`component::List`] with non-empty [`component::List::favorites`], the following
+/// methods should be called:
+///  - [`set_favorites`] to specify order and grouping of [`Component`]s in favorites;
+///  - [`extend`] to specify IDs of [`Component`]s which should be retained from the initial
+///    [`component::Group`]s specified through [`set_favorites`] (allows filtering of favorites e.g.
+///    by self type);
+///  - [`build`] to get a [`component::group::List`] of favorites constrained by arguments passed to
+///    the methods listed above.
```

Comment:
Again, this is not understandable at all. What are "favourites constrained by arguments"? I did not see any mention of constraints earlier. Why the call to "build" is even needed? It's not explained here. Please make docs understandable and very clear.